### ISSUE TITLE
grpo断点续训的critic/mean/score骤降

### ISSUE THREAD

**Issue by Yu7-code** [verl-project/verl#3754]
### System Info

你好！就是我断点续训后发现critic/score/mean出现一个骤降，请问是什么问题？附上我保存的checkpoint以及对应的wandb曲线

<img width="682" height="296" alt="Image" src="https://github.com/user-attachments/assets/92d21026-85d9-47b8-a8b4-b455ec149521" />

<img width="949" height="447" alt="Image" src="https://github.com/user-attachments/assets/86f294b5-1340-45e9-a9bd-59f73d285254" />

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

export RAY_DEDUP_LOGS=0

export WANDB_API_KEY=cbb7366270a128dc85ce035eba3ffc400aa6d7a3

math_train_path=./data/math2/train.parquet
math_test_path=./data/math2/test.parquet
# aime2025_test_path=./data/aime2025/test.parquet
# amc23_test_path=./data/amc23/test.parquet

train_files="['$math_train_path']"
# test_files="['$math_test_path', '$aime2025_test_path', '$amc23_test_path']"
test_files="['$math_test_path']"
kl_coef=0.001
lr=1e-6
model_name=Qwen/Qwen2.5-Math-1.5B

python3 -m verl.trainer.main_ppo \
    algorithm.adv_estimator=grpo \
    data.train_files="$train_files" \
    data.val_files="$test_files" \
    data.train_batch_size=1024 \
    data.max_prompt_length=1024 \
    data.max_response_length=3072 \
    data.filter_overlong_prompts=True \
    data.truncation='error' \
    actor_rollout_ref.model.path=$model_name \
    actor_rollout_ref.actor.optim.lr=$lr \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.actor.use_dynamic_bsz=True \
    actor_rollout_ref.actor.ppo_max_token_len_per_gpu=12000 \
    actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=16000 \
    actor_rollout_ref.ref.log_prob_max_token_len_per_gpu=16000 \
    actor_rollout_ref.actor.ppo_mini_batch_size=256 \
    actor_rollout_ref.actor.use_kl_loss=True \
    actor_rollout_ref.actor.kl_loss_coef=$kl_coef \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \
    actor_rollout_ref.actor.fsdp_config.param_offload=False \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=False \
    actor_rollout_ref.rollout.enforce_eager=False \
    actor_rollout_ref.rollout.free_cache_engine=False \
    actor_rollout_ref.rollout.tensor_model_parallel_size=2 \
    actor_rollout_ref.rollout.name=vllm \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.7 \
    actor_rollout_ref.rollout.n=8 \
    actor_rollout_ref.ref.fsdp_config.param_offload=True \
    trainer.experiment_name="MATH-Qwen2.5-Math-1.5b-GRPO" \
    trainer.critic_warmup=0 \
    trainer.logger=['wandb'] \
    trainer.project_name='verl' \
    trainer.n_gpus_per_node=2 \
    ++trainer.val_before_train=False \
    trainer.nnodes=1 \
    trainer.save_freq=7 \
    trainer.test_freq=7 \
    trainer.total_epochs=20 $@

### Expected behavior

断点续训critic/score/mean与之前相差不应过大

**Comment by Micro-han** [comment]
I met the same problem, have you solved?

**Comment by Yu7-code** [comment]
> I met the same problem, have you solved?

你好，方便加个微信一起讨论下吗，y18881911072

**Comment by zhangxy-2019** [comment]
请问问题解决了吗？我也遇到了

**Comment by zyx0725** [comment]
有人解决这个问题吗？

**Comment by bug-to-share** [comment]
同遇到这个问题

**Comment by Kirrito-k423** [comment]
中文：

谢谢反馈。这个现象还不能仅凭当前信息判断为可关闭的问题，也不适合用超时规则关闭：本 issue 有多位用户复现，且目前没有维护者给出结论。

从公开代码和文档看，`critic/score/mean` 是 batch 内 `token_level_scores` 聚合出来的序列得分均值，不等同于 critic 模型权重本身的指标；GRPO 下如果续训后的数据顺序、rollout 随机性、checkpoint 内容或恢复路径不一致，曲线可能出现跳变。verl 的 checkpoint 文档建议 FSDP 保存/加载完整训练状态（`model`、`optimizer`、`extra`），其中 `extra` 包含 LR scheduler 和 RNG 状态；trainer 代码也会在 checkpoint 下保存/恢复 `data.pt` dataloader 状态。

为了继续判断是否是 checkpoint/resume bug，请补充：

1. 使用的 verl 版本或 commit hash。
2. 实际续训命令，尤其是 `trainer.resume_mode` 和 `trainer.resume_from_path`。
3. checkpoint 目录结构，确认是否包含 actor/critic 的 model、optimizer、extra，以及 `global_step_xxx/data.pt`。
4. 续训启动日志中 `Load from checkpoint folder: ...` 附近的日志。
5. 断点前后同一 run 的 global step、`critic/score/mean`、`critic/rewards/mean`、`response_length/mean`、reward function 输出统计或 wandb 链接/截图。

English:

Thanks for the report. Based on the current public evidence, this should not be closed yet, and it should not be closed by the timeout rule: several users reported the same symptom, and there is no maintainer conclusion yet.

From the public code and docs, `critic/score/mean` is the mean sequence score aggregated from `token_level_scores` in the current batch; it is not itself a direct metric of the critic model weights. In GRPO, a jump after resume can happen if the resumed data order, rollout randomness, checkpoint contents, or resume path differs. The verl checkpoint docs recommend saving/loading the full FSDP training state (`model`, `optimizer`, and `extra`), where `extra` includes LR scheduler and RNG state; the trainer code also saves/restores dataloader state as `data.pt` under the checkpoint step directory.

To determine whether this is a checkpoint/resume bug, could you please provide:

1. The verl version or commit hash.
2. The exact resume command, especially `trainer.resume_mode` and `trainer.resume_from_path`.
3. The checkpoint directory layout, confirming whether actor/critic model, optimizer, extra states, and `global_step_xxx/data.pt` are present.
4. Logs around `Load from checkpoint folder: ...` during resume startup.
5. The global step before/after resume plus `critic/score/mean`, `critic/rewards/mean`, `response_length/mean`, reward-function output stats, or the relevant wandb link/screenshots.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
